tests/FloatTests.elm

```diff
@@ -0,0 +1,279 @@
+                    , Float.Extra.range 0 0 -1
+                        |> Expect.equal []
+                    , Float.Extra.range 0 0 -0.5
+                        |> Expect.equal []
+                    , Float.Extra.range 10 20 -2
+                        |> Expect.equal []
+                    , Float.Extra.range 10 20 -1
+                        |> Expect.equal []
+                    , Float.Extra.range 10 20 -0.5
+                        |> Expect.equal []
+                    ]
+        , test "returns an empty array if step is zero" <|
+            \() ->
+                Float.Extra.range 0 5 0
+                    |> Expect.equal []
+        , fuzz2 (Fuzz.intRange -1000 1000) (Fuzz.intRange -1000 1000) "behaves the same as List.range for ints" <|
+            \a b ->
+                Float.Extra.range (toFloat (min a b)) (toFloat (max a b)) 1
+                    |> List.map round
+                    -- account for List.range being inclusive
+                    |> (\y -> List.append y [ max a b ])
+                    |> Expect.equal (List.range (min a b) (max a b))
+        , test "returns exactly [start + step * i, …] for fractional steps" <|
+            \() ->
+                expectAll
+                    [ Float.Extra.range 0 0.5 0.1
+                        |> Expect.equal [ 0, 0.1, 0.1 * 2, 0.1 * 3, 0.1 * 4 ]
+                    , Float.Extra.range 0.5 0 -0.1
+                        |> Expect.equal [ 0.5, 0.5 - 0.1, 0.5 - 0.1 * 2, 0.5 - 0.1 * 3, 0.5 - 0.1 * 4 ]
+                    , Float.Extra.range -2 -1.2 0.1
+                        |> Expect.equal [ -2, -2 + 0.1, -2 + 0.1 * 2, -2 + 0.1 * 3, -2 + 0.1 * 4, -2 + 0.1 * 5, -2 + 0.1 * 6, -2 + 0.1 * 7 ]
+                    , Float.Extra.range -1.2 -2 -0.1
+                        |> Expect.equal [ -1.2, -1.2 - 0.1, -1.2 - 0.1 * 2, -1.2 - 0.1 * 3, -1.2 - 0.1 * 4, -1.2 - 0.1 * 5, -1.2 - 0.1 * 6, -1.2 - 0.1 * 7 ]
+                    ]
+        , test "returns exactly [start + step * i, …] for very small fractional steps" <|
+            \() ->
+                expectAll
+                    [ Float.Extra.range 2.1e-31 5.0e-31 1.1e-31
+                        |> Expect.equal [ 2.1e-31, 2.1e-31 + 1.1e-31, 2.1e-31 + 1.1e-31 * 2 ]
+                    , Float.Extra.range 5.0e-31 2.1e-31 -1.1e-31
+                        |> Expect.equal [ 5.0e-31, 5.0e-31 - 1.1e-31, 5.0e-31 - 1.1e-31 * 2 ]
+                    ]
+        , test "returns exactly [start + step * i, …] for very large fractional steps" <|
+            \() ->
+                expectAll
+                    [ Float.Extra.range 1.0e300 2.0e300 3.0e299
+                        |> Expect.equal [ 1.0e300, 1.0e300 + 3.0e299, 1.0e300 + 3.0e299 * 2, 1.0e300 + 3.0e299 * 3 ]
+                    , Float.Extra.range 2.0e300 1.0e300 -3.0e299
+                        |> Expect.equal [ 2.0e300, 2.0e300 - 3.0e299, 2.0e300 - 3.0e299 * 2, 2.0e300 - 3.0e299 * 3 ]
+                    ]
+        ]
```

Comment:
I don't think you do always have `abs step` elements? Step is a float, so that doesn't even make much sense, right?

Reply:
Uh, yeah, you're absolutely right, I mixed it up with `n`.

I can still think of a couple fuzz tests:
1. check that `abs (List.last - end) <= step`
2. check that `List.head == start`
3. check that `getAt (i+1) result - getAt i result ~= step`

I expect those to pass, but one thing I appreciate about fuzz testing is that it can find  corner cases you didn't expect

Reply:
I added the `List.head == start` one. The rest is just an exercise in getting the fuzzers to come up with cases that exercise floating point imprecision (and `range` being iterative is _quite_ subject to floating point drift).